### PR TITLE
(Hopefully) Reverts TG's Cloning Removal and adds Synthflesh cloning as a mid-game tech

### PR DESCRIPTION
**Pull request by SgtHunk**
## About The Pull Request

This reverts what the Cloning removal did for the most part, and adds Beestation's Synthflesh Cloning. Cloning will now be locked behind research, so engineers along with medical (With the help of chemists) will have to cooperate to build a functional Cloning.

## Why It's Good For The Game

Sam wanted this. Along with bringing more ways to get someone back into the game even if they're just a brain. 

## Changelog
:cl:
add: Reverted Cloning Removal
add: Ported Synthflesh cloning from Beestation
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
speedmerge this man

**Comment by SgtHunk** [comment]
Cloning revert speedrun any%

**Comment by Samuuu3** [comment]
Chad

**Comment by GoldenAlpharex** [comment]
Just to be sure, it clones your entire body, not just the brain, right?

File: code/game/machinery/dna_scanner.dm [hunk 1]
```diff
@@ -146,24 +146,3 @@
...
-	icon_state = "datadisk[rand(0,6)]"
-	add_overlay("datadisk_gene")
-
-/obj/item/disk/data/attack_self(mob/user)
-	read_only = !read_only
-	to_chat(user, "<span class='notice'>You flip the write-protect tab to [read_only ? "protected" : "unprotected"].</span>")
-
-/obj/item/disk/data/examine(mob/user)
-	. = ..()
-	. += "The write-protect tab is set to [read_only ? "protected" : "unprotected"]."
```
Comment (GoldenAlpharex) [inline code/game/machinery/dna_scanner.dm]: Why exactly are you removing the DNA disks? Are you sure this won't mess with Genetics?

File: code/game/objects/items/circuitboards/machine_circuitboards.dm [hunk 2]
```diff
@@ -910,8 +910,8 @@
 		/obj/item/stack/sheet/glass = 1)
 	def_components = list(/obj/item/stack/ore/bluespace_crystal = /obj/item/stack/ore/bluespace_crystal/artificial)
 
-/obj/item/circuitboard/machine/dnascanner
-	name = "DNA Scanner (Machine Board)"
+/obj/item/circuitboard/machine/clonescanner
+	name = "Cloning Scanner (Machine Board)"
```
Comment (GoldenAlpharex) [inline code/game/objects/items/circuitboards/machine_circuitboards.dm:914]: I'm curious, what's the reason behind renaming this? Did you check to see if it wouldn't tamper with Genetics?

Comment (GuillaumePrata) [reply code/game/objects/items/circuitboards/machine_circuitboards.dm:914]: I assume the clone scanner is just a improved version of the dna scanner, still probably better to keep both version or to swap the code from the clone scanner inside the dna scanner.
Having 2 machines still is the easiest way to go IMO.

**Comment by E231kei500bandai** [comment]
Cloning was removed from TG because it was bad for roleplay. Why would we remove it on fulpstation, an MRP server. The reason it was removed to begin with was why do anything as as doctor when you can just throw them in cryo if alive, cloning if dead. It just makes doctor incredibly boring and there basically isn't a point. Plus with the TG changes to chem, using chems like libital or aiuri is WORSE than just putting them in cryo or just letting them die in crit to clone them.

**Comment by SgtHunk** [comment]
Downstream moment

**Comment by AdvilPMSevere** [comment]
Based levels off the charts

**Comment by ilikesaying** [comment]
> Cloning was removed from TG because it was bad for roleplay. Why would we remove it on fulpstation, an MRP server. The reason it was removed to begin with was why do anything as as doctor when you can just throw them in cryo if alive, cloning if dead. It just makes doctor incredibly boring and there basically isn't a point. Plus with the TG changes to chem, using chems like libital or aiuri is WORSE than just putting them in cryo or just letting them die in crit to clone them.

yes but now it'll require science to research cloning, engineers to build the machines and chemistry to make or automate synthflesh, locking it behind TIME

**Comment by SgtHunk** [comment]
If Cloning still feels too early, I can lock it behind more research or make it more expensive to research, if that makes you comfortable

**Comment by E231kei500bandai** [comment]
fulp moment they test merged it before checking to see if it worked AT ALL like what the fuck

**Comment by SgtHunk** [comment]
Blame Salchems, already fixed on last commit

**Comment by E231kei500bandai** [comment]
still doesnt work still says Cannot start cloning: Not enough synthflesh. even though there is synthflesh

**Comment by SgtHunk** [comment]
Last commit must not be active yet then, I did message a mod and they said that the tgs did not show any updates regarding it, so you're still stuck on the 2nd commit

**Comment by Kholdgrave** [comment]
Fantastic. Nice to see that someone took the advice of the majority of players and put cloning back in. I'm legit excited about seeing a more efficient medbay and players not clearing out due to a lack of revival.

**Comment by SgtHunk** [comment]
aAaAaaaaaa undefined proc messes up UI, thank you proc very cool. Will fix it ASAP

**Comment by remove32** [comment]
> 
> 
> fulp moment they test merged it before checking to see if it worked AT ALL like what the fuck

what do you think a testmerge is for

**Comment by ATH1909** [comment]
> Cloning was removed from TG because it was bad for roleplay.

https://giphy.com/gifs/just-blinking-3ELtfmA4Apkju

> The reason it was removed to begin with was why do anything as as doctor when you can just throw them in cryo if alive, cloning if dead. It just makes doctor incredibly boring and there basically isn't a point. 

I was gonna respond to that first sentence, but you said what I would have said for me.

> Plus with the TG changes to chem, using chems like libital or aiuri is WORSE than just putting them in cryo or just letting them die in crit to clone them.

Actually, due to repeated misuse by doctors, the organ damage of libital, aiuri, etc. got watered down to the point where it's basically negligible now. The problem was less with cryo and more that cloning was faster, easier, and more multitask-able than literally every other method of reviving people (short of staves of healing and the like). It was at the point where if you spent the time to heal up a corpse and defib it instead of just cloning it, the person who you rezzed would get MAD at you for wasting their time (and yours).

File: code/datums/brain_damage/phobia.dm [hunk 3]
```diff
@@ -31,6 +31,11 @@
 	trigger_species = SStraumas.phobia_species[phobia_type]
 	..()
 
+
+/datum/brain_trauma/mild/phobia/on_clone()
+	if(clonable)
```
Comment (ATH1909) [inline code/datums/brain_damage/phobia.dm:36]: *clone**a**ble

**Comment by BootlegBow** [comment]
putting it behind research really fixes the issue of doctors just cloning instead of something else, although it might make advanced tech like limb growers and advanced tend wounds obsolete

**Comment by Kholdgrave** [comment]
Personally, I feel like limb growers and organ harvesters never see any real use. People usually print the Cybernetic organs and go to Robotics for replacement limbs. Advanced Tend Wounds would still see use since even when Cloners were implemented in their original state, there would be piles of corpses that had to wait in a queue, so people can just repair and revive while cloners do their thing. It'll be especially useful to have cloners during Station Wars like Ops or Blob, where the crew can be bled out through sheer attrition since the doctors can't possibly keep up with all the bodies, if the doctors are even alive and medbay is in tact.

**Pull request closed without merging**